AutomatedTesting/Gem/PythonTests/scripting/FileMenu_Default_NewAndOpen.py

```diff
@@ -50,38 +51,43 @@ class TestFileMenuDefaultNewOpen:
 
     @pyside_utils.wrap_async
     async def run_test(self):
-        # 1) Open Script Canvas window (Tools > Script Canvas)
+
+        # Preconditions
+        general.idle_enable(True)
+
+        # 1) Open Script Canvas window and wait for it to render(Tools > Script Canvas)
         general.open_pane("Script Canvas")
+        helper.wait_for_condition(lambda: general.is_pane_visible("Script Canvas"), TIME_TO_WAIT)
 
         # 2) Get the SC window object
         editor_window = pyside_utils.get_editor_main_window()
         sc = editor_window.findChild(QtWidgets.QDockWidget, "Script Canvas")
         sc_main = sc.findChild(QtWidgets.QMainWindow)
         sc_tabs = sc_main.findChild(QtWidgets.QTabWidget, "ScriptCanvasTabs")
 
-        # wait for the intial tab count
-        general.idle_wait(GENERAL_WAIT)
+        # 3) Get the initial tab count and verify it's zero. Save the tab count for verification in step 5
+        result = helper.wait_for_condition(lambda: sc_tabs.count() == 0, TIME_TO_WAIT)
```

Comment:
you don't need a wait_for_condition here the assumption is initial state is zero tabs and nothing is going to change that during this timeout
you can just put {sc_tabs.count() == 0} in the place of {result} in your Report.info